data-avro/src/main/java/com/linkedin/data/avro/DefaultAvroToDataConvertCallback.java

```diff
@@ -75,7 +75,7 @@ protected Object translateField(List<Object> path, Object fieldValue, RecordData
     DataSchema fieldDataSchema = field.getType();
     boolean isOptional = field.getOptional();
     Object result;
-    if (isOptional && fieldValue == Data.NULL)
+    if (isOptional && ((fieldValue == null) || (fieldValue == Data.NULL)))
```

Comment:
I don't think you need the inner parentheses

Reply:
I addressed both 1. and 2.

Reply:
For some reason though, the PR is not being updated with the remote branch.